lib/config/fileconf.go

```diff
@@ -250,10 +251,12 @@ type Log struct {
 	// Output defines where logs go. It can be one of the following: "stderr", "stdout" or
 	// a path to a log file
 	Output string `yaml:"output,omitempty"`
-	// Severity defines how verbose the log will be. Possible valus are "error", "info", "warn"
+	// Severity defines how verbose the log will be. Possible values are "error", "info", "warn"
 	Severity string `yaml:"severity,omitempty"`
 	// Format lists the output fields from KnownFormatFields. Example format: [timestamp, component, caller]
 	Format []string `yaml:"format,omitempty"`
```

Comment:
We can avoid adding extra field if we add this:

format: [json, timestamp, component, caller] // will activate json formatter
format: [fields, component, caller] // will activate fields formatter
format: [timestamp, component, caller] // defaults to fields

Can we also update docs as well?

Reply:
Can I ask why we'd like to avoid the new flag. In my opinion it's mixing the two is more confusing.

Also  the current implementation ignores the format field for the `json` formatter. `trace.JSONFormatter` seems to include the fields above by default but if we want the flexibility of choosing what we want we'd have to make a new modified version of the json formatter just like what's been done for the text formatter.

Reply:
Does the formatter always has to be at index 0 or can it be anywhere in the slice...

Also the way it's currently implemented, the fields are parsed after the formatter has been chosen. So we'd have to preparse the slice, remove the formatter from it, choose the formatter and then have the formatter finally parse it.

Reply:
Just to helicopter in @klizhentas but to me it doesn't seem very intuitive to embed the formatter in the existing Format field. Like `[json, timestamp, component, caller]` this doesn't mean anything to me, is it a combo of json and fields, where it's produced in json but then only includes the specified fields as a filter?

Reply:
Sure. Agree that adding json to the list is not the best option. Let's explore others. Two reasons why I'm trying to avoid extra flag:

* Locality. With `format: json`, we have all format-related sections in one key. With FormatType there is a second key that covers the format adjacent to the other one.
* Amount of permutations. The `FormatType: json` adds both key and a value, while Format: json only adds one value, so there is one less key to process/configure/validate.

may be we express format using data type?

// switches to json
`format: {timestamp: , component: , caller: }`

// uses text
`format: [timestamp, component, caller]`

on a second look, the  above probably should have been to be more consistent with this model:

`format: "{{timestamp}} {{component}} {{caller}}"`

Reply:
How about this:

```yaml
format:
   mode: json | template | rfc2379 (syslog)
   template: "{{timestamp}} {{component}} {{caller}} {{rest}}"
```

Reply:
I would consider `template -> go-template` (just in case we want to introduce x new format language)

For reference, kubectl options which is what I have in my mind as a similar type of formatter.
```
      --template='': Template string or path to template file to use when -o=go-template, -o=go-template-file. The
template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview].

  -o, --output='': Output format. One of:
json|yaml|wide|name|custom-columns=...|custom-columns-file=...|go-template=...|go-template-file=...|jsonpath=...|jsonpath-file=...
See custom columns [http://kubernetes.io/docs/user-guide/kubectl-overview/#custom-columns], golang template
[http://golang.org/pkg/text/template/#pkg-overview] and jsonpath template
[http://kubernetes.io/docs/user-guide/jsonpath].
```

Reply:
just to be clear the new config will look like the next two examples right now

```
format:
   output: text
   extra_fields: [timestamp, component, caller]  // these fields are added to whatever was going to be logged. 
```
The absence of the `fields` field, all the supported custom fields will be added by default, at least that's the current implementation.


```
format:
   output: json
   extra_fields: [timestamp, component, caller] // these fields are added to whatever was going to be logged. 
```
Same as above the new json output will add all the fields in the absence of the `fields` field.


That will allow the support of templates in the future. example
```
format:
   output: go-template
   extra_fields: [timestamp, component, caller]
   template: "{{.Timestamp}} {{.Component}} {{.Caller}} {{.Message}} ...."
```